src/Raku/ast/type.rakumod

```diff
@@ -311,6 +311,226 @@ class RakuAST::Type::Parameterized
+        QAST::Want.new(
+            $qast,
+            'v',
+            QAST::Op.new(:op('null'))
+        )
+    }
+
+    method PRODUCE-IMPLICIT-LOOKUPS() {
+        self.IMPL-WRAP-LIST([
+            RakuAST::Type::Setting.new(RakuAST::Name.from-identifier('Pair')),
+            RakuAST::Type::Setting.new(RakuAST::Name.from-identifier('List')),
+            RakuAST::Type::Setting.new(RakuAST::Name.from-identifier('Stringy')),
+            RakuAST::Type::Setting.new(RakuAST::Name.from-identifier('Numeric'))
+        ])
+    }
+
+    method IMPL-GENERATE-LEXICAL-DECLARATION(RakuAST::Name $name, Mu $type-object) {
+        RakuAST::VarDeclaration::Implicit::Constant.new:
+            :name($name),
+            :value($type-object),
+            :scope(self.scope);
+    }
+
+    method PERFORM-BEGIN(RakuAST::Resolver $resolver, RakuAST::IMPL::QASTContext $context) {
+        my @lookups      := self.IMPL-UNWRAP-LIST(self.get-implicit-lookups);
+        my $pair-type    := @lookups[0].resolution.compile-time-value;
+        my $list-type    := @lookups[1].resolution.compile-time-value;
+        my $stringy-type := @lookups[2].resolution.compile-time-value;
+        my $numeric-type := @lookups[3].resolution.compile-time-value;
+
+        my $base-type;
+        my $has-base-type := False;
+        if $!of {
+            $base-type := $!of.compile-time-value;
+            $has-base-type := True;
+        }
+        my %values := nqp::hash;
+        my $cur-val := nqp::box_i(-1, Int); # Boxed to support .succ
+        my $evaluated := self.IMPL-BEGIN-TIME-EVALUATE($!term, $resolver, $context);
+        my $is-settings-list := nqp::istype($evaluated, $list-type);
+        if nqp::istype($evaluated, $pair-type) {
+            if !$has-base-type {
+                # No need for type checking when we are going to get the base-type from the value
+                %values{$evaluated.key} := $evaluated.value;
+                $base-type := $evaluated.value.WHAT;
+            } else {
+                unless nqp::istype($evaluated.value, $!base-type) {
+                    nqp::die("Incorrect value type provided. Expected '" ~ $!base-type.raku ~ "' but got '" ~ $evaluated.value.WHAT.raku ~ "'");
+                }
+                %values{$evaluated.key} := $evaluated.value;
+            }
+        } elsif nqp::istype($evaluated, Str) {
+            # TODO: What do we actually want to do when base-type is defined but they only provide a single Str?
+            #       Base just ignores and uses Int
+            # A single string enum will always have 0, but we use $cur-val to keep it boxed
+            %values{$evaluated} := $cur-val.succ;
+            $base-type := Int;
+        } elsif nqp::istype($evaluated, List) || $is-settings-list {
+            my @items := $is-settings-list
+                            ?? $evaluated.FLATTENABLE_LIST
+                            !! self.IMPL-UNWRAP-LIST($evaluated);
```

Comment:
This distinction seems unnecessary as IMPL-UNWRAP-LIST is precicely for getting an NQP list out of a List object.

Reply:
Not sure what's going wrong then, but without this there was no iterating over `$evaluated` when a settings-level List object is returned from BEGIN-TIME-EVALUATE.

Reply:
You are correct, it works fine without it. Thanks for the clarification.